internal/runners/manifest/requirements.go

```diff
@@ -0,0 +1,146 @@
+package manifest
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/pkg/platform/api/buildplanner/model"
+	vulnModel "github.com/ActiveState/cli/pkg/platform/api/vulnerabilities/model"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	platformModel "github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+)
+
+type requirement struct {
+	NameOutput      string `json:"name" locale:"manifest_name,Name"`
+	VersionOutput   string `json:"version" locale:"manifest_version,Version"`
+	License         string `json:"license" locale:"manifest_license,License"`
+	Vulnerabilities string `json:"vulnerabilities" locale:"manifest_vulnerabilities,Vulnerabilities (CVEs)" opts:"omitEmpty"`
+	// Must be last of the output fields in order for our table renderer to include all the fields before it
+	NamespaceOutput string `json:"namespace" locale:"manifest_namespace,Namespace" opts:"omitEmpty,separateLine"`
+
+	// These fields are used for internal processing
+	name      string
+	namespace string
+}
+
+type requirementsOutput struct {
+	Requirements []*requirement `json:"requirements"`
+}
+
+func newRequirementsOutput(reqs []model.Requirement, artifacts []*artifact.Artifact, vulns vulns, auth *authentication.Auth) (requirementsOutput, error) {
+	var requirements []*requirement
+	for _, req := range reqs {
+		r := &requirement{
+			NameOutput: locale.Tl("manifest_name", "[ACTIONABLE]{{.V0}}[/RESET]", req.Name),
+			namespace:  req.Namespace,
+			name:       req.Name,
+		}
+
+		var version string
+		if req.VersionRequirement != nil {
+			version = locale.Tl("manifest_constraint_resolved", "[CYAN]{{.V0}}[/RESET]", platformModel.BuildPlannerVersionConstraintsToString(req.VersionRequirement))
+		} else {
+			version = locale.Tl("manifest_constraint_auto", "[CYAN]auto[/RESET]")
+			for _, a := range artifacts {
+				if a.Namespace == req.Namespace && a.Name == req.Name {
+					version = locale.Tl("manifest_constraint_resolved", "[CYAN]{{.V0}}[/RESET] → [CYAN]{{.V1}}[/RESET]", version, *a.Version)
+					break
+				}
+			}
+		}
+		r.VersionOutput = version
+
+		normalized, err := platformModel.FetchNormalizedName(req.Namespace, req.Name, auth)
+		if err != nil {
+			multilog.Error("Failed to normalize '%s': %v", req.Name, err)
+		}
+
+		packages, err := platformModel.SearchIngredientsStrict(req.Namespace, normalized, false, false, nil, auth)
+		if err != nil {
+			multilog.Error("Failed to search for '%s': %v", req.Name, err)
+		}
```

Comment:
I did not realize the license info was not part of the buildplan, but that makes sense in retrospect. If we cannot retrieve this license info in bulk then we should drop this. It's nice to have but not to the point that we want to be spamming our API.

Note any time you are calling an API inside a loop alarm bells should be going off. We should always try to avoid this.